_posts/2025-05-08-opensearch-performance-3.0.md

```diff
@@ -0,0 +1,1013 @@
+* **Term aggregations**: Latency dropped from ~609 ms in 1.3 to ~80 ms in 3.0 (**~85% faster**, **~30% compared to 2.17**). Improvements include optimized global ordinals, smarter doc values access, and execution hints for controlling cardinality aggregation memory usage ([#17657](https://github.com/opensearch-project/OpenSearch/pull/17657)). Approximate filtering for range constraints also reduced scan costs in observability use cases.
+* **Sort queries**: Sorted queries are **~60% faster** than 1.3 and **~5--10% faster** than 2.17. OpenSearch 3.0 restored Lucene's time-series sort optimization by correctly invoking the new `searchAfterPartition()` API ([PR #17329](https://github.com/opensearch-project/OpenSearch/pull/17329)), resolving regressions in `desc_sort_timestamp` queries ([#17404](https://github.com/opensearch-project/OpenSearch/issues/17404)).
+* **Date histograms**: Latency fell from ~6s in 1.3 to ~146 ms in 3.0, with lows of ~85 ms using concurrency (up to **60x faster**). Gains come from block skipping, segment-level parallelism, and [sub-agg optimizations](https://github.com/opensearch-project/OpenSearch/pull/17447) (for example, pushing filters inside range-auto-date aggs). 3.0 is up to **~50% faster than 2.19** in this category.
+* **Range queries**: Range filters now run in ~2.75 ms, down from ~26 ms in 1.3 (**~89% faster**) and **~25% faster than 2.19**. OpenSearch's [range approximation framework](https://github.com/opensearch-project/OpenSearch/issues/16682) became generally available in 3.0, delivering gains through more efficient use of doc values and by skipping over non-matching blocks, further lowering scan costs.
+
+### Star-tree index for log analytics
+
+Introduced in 2.19, the **[star-tree aggregator](https://github.com/opensearch-project/opensearch-build/blob/main/release-notes/opensearch-release-notes-2.19.0.md)** enables pre-aggregated metric and filtered terms queries. In eligible use cases, this reduces query compute by up to **100x** and cache usage by **~30x**. 3.0 extends support to multi-level and numeric terms aggregations (with [sorted bucket elimination](https://github.com/opensearch-project/OpenSearch/pull/17671)), achieving **~15–20% lower latency** over 2.19 in benchmarks.
+
+### Concurrent segment search (auto mode)
+
+OpenSearch 3.0 defaults to **concurrent segment search in the `auto` mode**, allowing the engine to choose between sequential and parallel segment execution per query. Expensive aggregations (for example, `terms` or `date_histogram`) run in parallel, while lightweight queries stay single-threaded to avoid overhead. On an 8-vCPU cluster, this yielded **~35% lower latency on date histograms**, **15%+ on terms aggregations**, and showed an **~8% aggregate gain** across the Big 5. Text queries stayed flat (for example, 8.22 ms sequential compared to 8.93 ms auto), confirming the selector works as intended without regression.
```

Comment:
```suggestion
OpenSearch 3.0 defaults to **concurrent segment search in the `auto` mode**, allowing the engine to choose between sequential and parallel segment execution per query. Expensive aggregations (for example, `terms` or `date_histogram`) run in parallel, while lightweight queries stay single-threaded to avoid overhead. On an 8-vCPU cluster, this yielded **~35% lower latency on date histograms**, **15%+ on terms aggregations**, and showed an **~8% aggregate gain** across the Big5. Text queries stayed flat (for example, 8.22 ms sequential compared to 8.93 ms auto), confirming that the selector works as intended without regression.
```